### PR TITLE
OLS-1055: switch to gpt-4o-mini

### PR DESCRIPTION
**Pull request by onmete**
## Description

switch to gpt-4o-mini

## Type of change

- [x] New feature

### CODE REVIEW

File: tests/e2e/test_api.py [hunk 1]
```diff
@@ -790,9 +786,8 @@ def test_query_filter() -> None:
         response_text = json_response["response"].lower()
         assert "openshift" in response_text
         assert "deploy" in response_text
-        response_words = response_text.split()
-        assert "foo" not in response_words
-        assert "bar" not in response_words
+        assert "foo" not in response_text
```
Comment (tisnik) [inline tests/e2e/test_api.py:789]: the split was there on purpose, especially because "bar" is part of regular words

**Review by tisnik — CHANGES_REQUESTED** [review]

File: tests/e2e/test_api.py (hunk 1 above)
Comment (onmete) [reply tests/e2e/test_api.py]: I don't get it
```
>>> l = "bu ba bo"
>>> "be" not in l
True
>>> "be" not in l.split()
True
```
but ok.

File: ols/src/prompts/prompts.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 If the context of the question is not clear, consider it to be OpenShift.
 Never include URLs in your replies.
 Refuse to answer questions or execute commands not about OpenShift.
+If asked, you provide information about Kubernetes as well.
```
[the remark this answers is not in the document]
Comment (onmete) [reply ols/src/prompts/prompts.py:27]: yes, the eval1 - what is kubernetes failed with llm saying it is not question about openshift

File: scripts/evaluation/eval_data/question_answer_pair.json [hunk 3]
```diff
@@ -276,15 +277,15 @@
...
+                        "To find your cluster ID in OpenShift, you can use either the web console or the OpenShift CLI (oc). Here are the steps for both methods:Using the web console:1. Log in to the OpenShift web console.2. Navigate to Home -> Overview.3. In the Details section, you will find your cluster ID displayed in the Cluster ID field.Using the OpenShift CLI (oc):1. Log in to your cluster using the OpenShift CLI (oc) with a user account that has cluster-admin role access.2. Run the following command:   ```   $ oc get clusterversion -o jsonpath='{.items[].spec.clusterID}{}'   ```   This command will retrieve and display your cluster ID.By following these steps, you should be able to find your cluster ID in OpenShift."
```
[the remark this answers is not in the document]
Comment (onmete) [reply scripts/evaluation/eval_data/question_answer_pair.json:288]: yes, this must come from that change in the system instructions

File: tests/e2e/test_api.py [hunk 4]
```diff
@@ -770,6 +765,7 @@ def test_rag_question() -> None:
         assert len(doc_urls_list) == len(set(doc_urls_list))
 
 
+@pytest.mark.cluster()
```
[the remark this answers is not in the document]
Comment (onmete) [reply tests/e2e/test_api.py:768]: not really as there are `cluster_utils` functions used, I would say local testing doesn't involve cluster, but just running instance of service

File: ols/src/prompts/prompts.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (onmete) [reply ols/src/prompts/prompts.py]: I was using rag locally, but for some reason I have different evaluation results than in CI. I've removed this instruction and it seems alright on the CI side (just different failures). So I'll remove this.

**Comment by tisnik** [comment]
/override "Red Hat Konflux / lightspeed-service-on-pull-request"
/override "Red Hat Konflux / ols-enterprise-contract / lightspeed-service"

**Pull request merged**